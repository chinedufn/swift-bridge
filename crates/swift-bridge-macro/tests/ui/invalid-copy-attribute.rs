//! # To Run
//! cargo test -p swift-bridge-macro -- ui trybuild=invalid-copy-attribute.rs

#[swift_bridge::bridge]
mod ffi {
    extern "Rust" {
        #[swift_bridge(Copy(1))]
        type DoesNotImplementCopy;

        #[swift_bridge(Copy(10))]
        type IncorrectCopySize;

    }
}

pub struct DoesNotImplementCopy(u8);

#[derive(Copy, Clone)]
pub struct IncorrectCopySize([u8; 9]);

fn main() {}
