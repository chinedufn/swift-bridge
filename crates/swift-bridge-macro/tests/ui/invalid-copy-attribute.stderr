error[E0308]: mismatched types
  --> tests/ui/invalid-copy-attribute.rs:11:14
   |
11 |         type IncorrectCopySize;
   |              ^^^^^^^^^^^^^^^^^
   |              |
   |              expected an array with a fixed size of 9 elements, found one with 10 elements
   |              expected due to this

error[E0277]: the trait bound `DoesNotImplementCopy: Copy` is not satisfied
  --> tests/ui/invalid-copy-attribute.rs:8:14
   |
8  |         type DoesNotImplementCopy;
   |              ^^^^^^^^^^^^^^^^^^^^ the trait `Copy` is not implemented for `DoesNotImplementCopy`
   |
note: required by a bound in `assert_copy`
  --> $WORKSPACE/src/lib.rs
   |
   | pub fn assert_copy<T: Copy>() {}
   |                       ^^^^ required by this bound in `assert_copy`
help: consider annotating `DoesNotImplementCopy` with `#[derive(Copy)]`
   |
16 | #[derive(Copy)]
   |
